Library/Homebrew/cmd/update-report.rb

```diff
@@ -104,6 +105,36 @@ def update_report
       next unless tap.git?
       next if tap.core_tap? && ENV["HOMEBREW_INSTALL_FROM_API"].present? && args.preinstall?
 
+      # HOMEBREW_CORE_MERGE_MAINTAINER can be set by the maintainers
+      # who merge homebrew-core into linuxbrew-core to avoid being migrated
+      # We only migrate HOMEBREW_DEVELOPER users right now
+      # Don't migrate on CI either
+      if ENV["HOMEBREW_DEVELOPER"].present? &&
+         ENV["HOMEBREW_CORE_MERGE_MAINTAINER"].blank? &&
+         ENV["GITHUB_ACTIONS"].blank? &&
+         tap.core_tap? &&
+         tap.remote == "https://github.com/Homebrew/linuxbrew-core"
+        puts_stdout_or_stderr "Migrate from linuxbrew-core to homebrew-core" unless args.quiet?
+
+        LINUXBREW_CORE_MIGRATION_LIST.each do |name|
+          formula = Formula[name]
+          next unless formula.any_version_installed?
+
+          selected_keg = nil
+          version = formula.installed_kegs.map(&:version).max
+          formula.installed_kegs.each do |keg|
+            selected_keg = keg if version == keg.version
+          end
+          tab = Tab.for_keg(selected_keg)
+          tab.version_scheme = tab.version_scheme - 1
+          tab.write
+        end
+
+        safe_system "git", "-C", tap.path, "remote", "set-url", "origin", "https://github.com/Homebrew/homebrew-core"
+        safe_system "git", "-C", tap.path, "fetch", "origin", "master"
+        safe_system "git", "-C", tap.path, "reset", "--hard", "origin/master"
```

Comment:
Given the requirements on the above here: I'm tempted to say this is something that should be:
1) either done before by `update.sh` (which sets an environment variable to do the formulae migration)
2) calls `update.sh` again which will do this migration (because it knows how to do all the e.g. stashing, remote changing, etc.)

Reply:
I agree that we should re-use `update.sh`'s code that stashes and pops changes if possible. I need to investigate how/where to move my code and how it fits in `update.sh`.

Reply:
The more I think about this I think 1) will be a bit cleaner here.

If `HOMEBREW_CORE_DEFAULT_GIT_REMOTE`  and `HOMEBREW_CORE_GIT_REMOTE` don't match then `update.sh` will automatically update the remote and fetch it accordingly. This could be done in `brew.sh` and then `update.sh` does something like `export HOMEBREW_LINUXBREW_MIGRATION=1` which will then trigger updating the `version_scheme` on those formulae in `update-report.rb`.

Reply:
The drawback is that if something fails between the repo change and the update-report migration step, the migration will never be completed. I would prefer to do the `version_scheme` migration first, and then change the remote after that.

But we might have no choice if we want to benefit from all the stash/pop capabilities of `update.sh`